registrar/apps/api/v1/tests/test_views.py

```diff
@@ -2444,6 +2444,98 @@ def build_enrollment(self, status, student_key=None, course_id=None):
         }
 
 
+class CourseEnrollmentUploadCourseStaffTest(EnrollmentUploadMixin, S3MockMixin, RegistrarAPITestCase, AuthRequestMixin):
+    """ Test course enrollment csv upload """
+    path = 'programs/masters-in-cs/course_enrollments/upload/'
+    event = 'registrar.v1.upload_course_enrollments'
+    csv_headers = ('student_key', 'course_id', 'status', 'course_staff')
+
+    def build_enrollment(self, status, student_key=None, course_staff=None, course_id=None):
+        return {
+            'status': status,
+            'student_key': student_key or uuid.uuid4().hex[0:10],
+            'course_id': course_id or uuid.uuid4().hex[0:10],
+            'course_staff': course_staff
+        }
+    
+    @mock.patch.object(ProgramEnrollmentUploadView, 'task_fn', _succeeding_job)
+    @mock.patch.object(CourseRunEnrollmentUploadView, 'task_fn', _succeeding_job)
+    def test_enrollment_upload_success(self):
+        enrollments = [
+            self.build_enrollment('enrolled', '001', 'True'),
+            self.build_enrollment('pending', '002', 'False'),
+            self.build_enrollment('enrolled', '003', 'True'),
+        ]
+
+        with self.assert_tracking(
+                user=self.stem_admin,
+                program_key=self.cs_program.key,
+                status_code=202
+        ):
+            with activate_waffle_flag('enable_course_role_management', self.stem_admin_group):
+                upload_response = self._upload_enrollments(enrollments)
+
+        self.assertEqual(upload_response.status_code, 202)
+
+        job_response = self.get(upload_response.data['job_url'], self.stem_admin)
+        self.assertEqual(job_response.status_code, 200)
+        self.assertEqual(job_response.data['state'], 'Succeeded')
+
+        filestore = get_enrollment_uploads_filestore()
+        retrieved = filestore.retrieve('/{}/{}.json'.format(
+            'uploads',
+            upload_response.data['job_id'],
+        ))
+        self.assertEqual(json.loads(retrieved), enrollments)
+
+    @mock.patch.object(ProgramEnrollmentUploadView, 'task_fn', _succeeding_job)
+    @mock.patch.object(CourseRunEnrollmentUploadView, 'task_fn', _succeeding_job)
+    @override_flag('enable_course_role_management', active=False)
+    def test_enrollment_upload_success_ignore_course_staff_when_waffle_disabled(self):
+        enrollments = [{'course_id': 'ac2102fb65', 'status': 'enrolled', 'student_key': '001', 'course_staff': 'True'},
+                       {'course_id': '12a8bf290c', 'status': 'pending', 'student_key': '002', 'course_staff': 'False'},
+                       {'course_id': 'e7826f975e', 'status': 'enrolled', 'student_key': '003', 'course_staff': 'True'}]
+
+        expected_enrollments = [{'course_id': 'ac2102fb65', 'status': 'enrolled', 'student_key': '001'},
```

Comment:
#NIT This array can be constructed based on the last `enrollments` array.

```suggestion
        expected_enrollments = [enrollment_item.pop('course_staff') for enrollment_item in enrollments]
```

Reply:
Won't that modify the items in `enrollments`?

Reply:
@kdmccormick Huh, I didn't realize that. But I think we can do copy then?
Frankly, I am also fine leaving this as it is.

Reply:
As-is strikes me as fine. Sometimes I like when tests have their data fully spelled-out instead of created by Python logic. The latter can obscure what the test is doing and hide bugs.